components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/LocalEntryResource.java

```diff
@@ -0,0 +1,179 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.micro.integrator.management.apis;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.config.Entry;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * This class provides mechanisms to monitor local entries deployed.
+ * */
+public class LocalEntryResource implements MiApiResource {
+
+    private static final Log LOG = LogFactory.getLog(LocalEntryResource.class);
+
+    // HTTP method types supported by the resource
+    Set<String> methods;
+    // Constants for local-entry json object
+    private static final String TYPE_ATTRIBUTE= "type";
+    private static final String VALUE_ATTRIBUTE= "value";
+    private static final String DESCRIPTION_ATTRIBUTE= "description";
+
+    public LocalEntryResource() {
+        methods = new HashSet<>();
+        methods.add(Constants.HTTP_GET);
+    }
+    @Override
+    public Set<String> getMethods() {
+        return methods;
+    }
+
+    @Override
+    public boolean invoke(MessageContext messageContext,
+                          org.apache.axis2.context.MessageContext axis2MessageContext,
+                          SynapseConfiguration synapseConfiguration) {
+
+        String entryKey = Utils.getQueryParameter(messageContext, Constants.NAME);
+        if (Objects.nonNull(entryKey)) {
+            populateLocalEntryData(axis2MessageContext, synapseConfiguration, entryKey);
+        } else {
+            populateLocalEntries(axis2MessageContext, synapseConfiguration);
+        }
+        axis2MessageContext.removeProperty(Constants.NO_ENTITY_BODY);
+        return true;
+    }
+
+    /**
+     * Sets the list of all available local entries to the response as json.
+     *
+     * @param axis2MessageContext axis2 message context
+     * @param synapseConfiguration Synapse configuration object
+     */
+    private void populateLocalEntries(org.apache.axis2.context.MessageContext axis2MessageContext,
+                                      SynapseConfiguration synapseConfiguration) {
+
+        Map definedEntries = synapseConfiguration.getDefinedEntries();
+        // (-2) to account for the 2 local entries defined by the server
+        JSONObject jsonBody = Utils.createJSONList(definedEntries.size() - 2);
+        for (Object o : definedEntries.entrySet()) {
```

Comment:
Can't we use for (Map.Entry<String,Entry> definedEntry : definedEntries.entrySet()) instead of this? This will avoid having to cast keys and values at lines, 100 and 107.